// @lobehub/ui/es/mdx/mdxComponents/Citation/index.js
'use client';

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
var _templateObject, _templateObject2;
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }
import { createStyles } from 'antd-style';
import { isEmpty } from 'lodash-es';
import PopoverPanel from "./PopoverPanel";
import { jsx as _jsx } from "react/jsx-runtime";
var useStyles = createStyles(function (_ref) {
  var css = _ref.css;
  return {
    container: css(_templateObject || (_templateObject = _taggedTemplateLiteral(["\n    display: inline-flex;\n    line-height: var(--lobe-markdown-line-height);\n    vertical-align: baseline;\n\n    a {\n      color: inherit;\n    }\n  "]))),
    supContainer: css(_templateObject2 || (_templateObject2 = _taggedTemplateLiteral(["\n    vertical-align: super;\n  "])))
  };
});
var Citation = function Citation(_ref2) {
  var children = _ref2.children,
    href = _ref2.href,
    inSup = _ref2.inSup,
    id = _ref2.id,
    citationDetail = _ref2.citationDetail;
  var _useStyles = useStyles(),
    styles = _useStyles.styles,
    cx = _useStyles.cx;
  var usePopover = !isEmpty(citationDetail);
  var url = (citationDetail === null || citationDetail === void 0 ? void 0 : citationDetail.url) || href;

  // [^1] 格式类型
  if (inSup) {
    return /*#__PURE__*/_jsx(PopoverPanel, _objectSpread(_objectSpread({}, citationDetail), {}, {
      usePopover: usePopover,
      children: /*#__PURE__*/_jsx("span", {
        className: styles.container,
        children: /*#__PURE__*/_jsx("a", {
          "aria-describedby": "footnote-label",
          "data-footnote-ref": "true",
          href: url,
          id: id,
          rel: "noreferrer",
          target: citationDetail !== null && citationDetail !== void 0 && citationDetail.url ? '_blank' : undefined,
          children: children
        })
      })
    }));
  }
  return /*#__PURE__*/_jsx(PopoverPanel, _objectSpread(_objectSpread({}, citationDetail), {}, {
    usePopover: usePopover,
    children: /*#__PURE__*/_jsx("sup", {
      className: cx(styles.container, styles.supContainer),
      children: url ? /*#__PURE__*/_jsx("a", {
        "aria-describedby": "footnote-label",
        "data-footnote-ref": true,
        href: url,
        rel: "noreferrer",
        target: '_blank',
        children: children
      }) : /*#__PURE__*/_jsx("span", {
        "aria-describedby": "footnote-label",
        "data-footnote-ref": true,
        children: children
      })
    })
  }));
};
Citation.displayName = 'MdxCitation';
export default Citation;
