// @lobehub/ui/es/Snippet/style.js
var _templateObject, _templateObject2;
function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }
import { createStyles } from 'antd-style';
export var useStyles = createStyles(function (_ref) {
  var css = _ref.css,
    token = _ref.token,
    stylish = _ref.stylish;
  return {
    borderless: stylish.variantBorderlessWithoutHover,
    filled: stylish.variantFilledWithoutHover,
    hightlight: css(_templateObject || (_templateObject = _taggedTemplateLiteral(["\n      overflow: auto hidden;\n      flex: 1;\n      height: 100%;\n      padding: 0;\n\n      pre {\n        display: flex;\n        align-items: center;\n        height: 100%;\n      }\n    "]))),
    outlined: stylish.variantOutlinedWithoutHover,
    root: css(_templateObject2 || (_templateObject2 = _taggedTemplateLiteral(["\n      position: relative;\n\n      overflow: hidden;\n\n      max-width: 100%;\n      height: 38px;\n      padding-block: 0;\n      padding-inline: 12px 8px;\n\n      border-radius: ", "px;\n    "])), token.borderRadius),
    shadow: stylish.shadow
  };
});
