// react-layout-kit/lib/utils.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.isVertical = exports.isSpaceDistribution = exports.isHorizontal = exports.getPrefix = exports.getFlexDirection = exports.getCssValue = void 0;
var getPrefix = exports.getPrefix = function getPrefix(prefixCls) {
  if (prefixCls) return prefixCls;
  return 'layoutkit';
};
var getFlexDirection = exports.getFlexDirection = function getFlexDirection(direction, isHorizontal) {
  if (isHorizontal) return 'row';
  switch (direction) {
    case 'horizontal':
      return 'row';
    case 'horizontal-reverse':
      return 'row-reverse';
    case 'vertical':
    default:
      return 'column';
    case 'vertical-reverse':
      return 'column-reverse';
  }
};
var isSpaceDistribution = exports.isSpaceDistribution = function isSpaceDistribution(distribution) {
  if (!distribution) return;
  return ['space-between', 'space-around', 'space-evenly'].includes(distribution);
};
var isHorizontal = exports.isHorizontal = function isHorizontal(direction, _isHorizontal) {
  return getFlexDirection(direction, _isHorizontal) === 'row';
};
var isVertical = exports.isVertical = function isVertical(direction, isHorizontal) {
  return getFlexDirection(direction, isHorizontal) === 'column';
};
var getCssValue = exports.getCssValue = function getCssValue(value) {
  return typeof value === 'number' ? "".concat(value, "px") : value;
};
