// @lobehub/ui/es/mobile/ChatInputArea/style.js
var _templateObject, _templateObject2, _templateObject3, _templateObject4, _templateObject5;
function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }
import { createStyles } from 'antd-style';
export var useStyles = createStyles(function (_ref) {
  var css = _ref.css,
    token = _ref.token;
  return {
    container: css(_templateObject || (_templateObject = _taggedTemplateLiteral(["\n      flex: none;\n      padding-block: 12px;\n      background: ", ";\n      border-block-start: 1px solid ", ";\n    "])), token.colorFillQuaternary, token.colorFillTertiary),
    expand: css(_templateObject2 || (_templateObject2 = _taggedTemplateLiteral(["\n      position: absolute;\n      width: 100%;\n      height: 100%;\n    "]))),
    expandButton: css(_templateObject3 || (_templateObject3 = _taggedTemplateLiteral(["\n      position: absolute;\n      inset-inline-end: 14px;\n    "]))),
    expandTextArea: css(_templateObject4 || (_templateObject4 = _taggedTemplateLiteral(["\n      flex: 1;\n    "]))),
    inner: css(_templateObject5 || (_templateObject5 = _taggedTemplateLiteral(["\n      height: inherit;\n      padding-block: 0;\n      padding-inline: 8px;\n    "])))
  };
});
