// @lobehub/ui/es/Segmented/style.js
var _templateObject, _templateObject2, _templateObject3;
function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }
import { createStyles } from 'antd-style';
export var useStyles = createStyles(function (_ref) {
  var css = _ref.css,
    stylish = _ref.stylish,
    token = _ref.token;
  return {
    borderless: stylish.variantBorderlessWithoutHover,
    filled: css(_templateObject || (_templateObject = _taggedTemplateLiteral(["\n      background: ", ";\n      border: 1px solid ", ";\n    "])), token.colorBgLayout, token.colorFillQuaternary),
    glass: stylish.blur,
    outlined: css(_templateObject2 || (_templateObject2 = _taggedTemplateLiteral(["\n      background: transparent;\n      border: 1px solid ", ";\n    "])), token.colorBorderSecondary),
    root: css(_templateObject3 || (_templateObject3 = _taggedTemplateLiteral([""]))),
    shadow: stylish.shadow
  };
});
