// @lobehub/ui/es/Avatar/style.js
var _templateObject, _templateObject2;
function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }
import { createStyles } from 'antd-style';
export var useStyles = createStyles(function (_ref) {
  var stylish = _ref.stylish,
    css = _ref.css,
    token = _ref.token,
    prefixCls = _ref.prefixCls;
  return {
    borderless: stylish.variantBorderlessWithoutHover,
    filled: stylish.variantFilledWithoutHover,
    loading: css(_templateObject || (_templateObject = _taggedTemplateLiteral(["\n      position: absolute;\n      color: #fff;\n      background: ", ";\n    "])), token.colorBgMask),
    outlined: stylish.variantOutlinedWithoutHover,
    root: css(_templateObject2 || (_templateObject2 = _taggedTemplateLiteral(["\n      background: transparent;\n      &.", "-avatar {\n        user-select: none;\n\n        overflow: hidden;\n        display: flex;\n        align-items: center;\n        justify-content: center;\n\n        border: none;\n\n        .", "-avatar-string {\n          transform: none !important;\n\n          overflow: hidden;\n          display: flex;\n          align-items: center;\n          justify-content: center;\n\n          width: 100%;\n          height: 100%;\n          padding: 0;\n\n          font-size: inherit;\n          font-weight: bolder;\n          line-height: 1;\n          color: inherit;\n        }\n      }\n    "])), prefixCls, prefixCls),
    shadow: stylish.shadow
  };
});
