// @lobehub/ui/es/awesome/Features/FeatureItem.js
'use client';

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
var _excluded = ["style", "className", "row", "column", "description", "image", "title", "link", "icon", "imageStyle", "openExternal"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import { memo } from 'react';
import { Center, Flexbox } from 'react-layout-kit';
import A from "../../A";
import Icon from "../../Icon";
import Img from "../../Img";
import Text from "../../Text";
import { useStyles } from "./style";
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
var Image = /*#__PURE__*/memo(function (_ref) {
  var image = _ref.image,
    className = _ref.className,
    title = _ref.title,
    style = _ref.style;
  return image.startsWith('http') ? /*#__PURE__*/_jsx(Img, {
    alt: title,
    className: className,
    src: image,
    style: style
  }) : /*#__PURE__*/_jsx(Center, {
    className: className,
    style: style,
    children: image
  });
});
var Item = /*#__PURE__*/memo(function (_ref2) {
  var style = _ref2.style,
    className = _ref2.className,
    row = _ref2.row,
    column = _ref2.column,
    description = _ref2.description,
    image = _ref2.image,
    title = _ref2.title,
    link = _ref2.link,
    icon = _ref2.icon,
    imageStyle = _ref2.imageStyle,
    openExternal = _ref2.openExternal,
    rest = _objectWithoutProperties(_ref2, _excluded);
  var rowNumber = row || 7;
  var _useStyles = useStyles({
      hasLink: Boolean(link),
      rowNum: rowNumber
    }),
    styles = _useStyles.styles,
    cx = _useStyles.cx;
  return /*#__PURE__*/_jsx("div", _objectSpread(_objectSpread({
    className: cx(styles.container, className),
    style: _objectSpread({
      gridColumn: "span ".concat(column || 1),
      gridRow: "span ".concat(rowNumber)
    }, style)
  }, rest), {}, {
    children: /*#__PURE__*/_jsxs("div", {
      className: styles.cell,
      children: [image || icon && /*#__PURE__*/_jsxs(Center, {
        className: styles.imgContainer,
        style: imageStyle,
        children: [icon && /*#__PURE__*/_jsx(Icon, {
          className: styles.img,
          icon: icon
        }), image && /*#__PURE__*/_jsx(Image, {
          className: styles.img,
          image: image,
          title: title
        })]
      }), title && /*#__PURE__*/_jsx(Flexbox, {
        align: 'center',
        as: 'h3',
        className: styles.title,
        gap: 8,
        horizontal: true,
        children: title
      }), description && /*#__PURE__*/_jsx(Text, {
        className: styles.desc,
        ellipsis: {
          rows: 4
        },
        children: description
      }), link && /*#__PURE__*/_jsx("div", {
        className: styles.link,
        children: /*#__PURE__*/_jsx(A, {
          href: link,
          rel: "noreferrer",
          target: openExternal ? '_blank' : undefined,
          children: "Read More"
        })
      })]
    })
  }));
});
Item.displayName = 'FeatureItem';
export default Item;
