// @lobehub/ui/es/Button/Button.js
'use client';

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
var _excluded = ["icon", "variant", "glass", "shadow", "loading", "className", "type", "color", "danger", "children", "iconProps", "ref"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import { Button as AntdButton } from 'antd';
import { cva } from 'class-variance-authority';
import { Loader2Icon } from 'lucide-react';
import { isValidElement, memo, useMemo } from 'react';
import Icon from "../Icon";
import { useStyles } from "./style";
import { jsx as _jsx } from "react/jsx-runtime";
var Button = /*#__PURE__*/memo(function (_ref) {
  var icon = _ref.icon,
    variant = _ref.variant,
    glass = _ref.glass,
    shadow = _ref.shadow,
    loading = _ref.loading,
    className = _ref.className,
    type = _ref.type,
    color = _ref.color,
    danger = _ref.danger,
    children = _ref.children,
    iconProps = _ref.iconProps,
    ref = _ref.ref,
    rest = _objectWithoutProperties(_ref, _excluded);
  var _useStyles = useStyles(),
    styles = _useStyles.styles,
    cx = _useStyles.cx,
    theme = _useStyles.theme;
  var defaultVariant = type ? undefined : variant || (theme.isDarkMode ? 'filled' : 'outlined');
  var variants = useMemo(function () {
    return cva(styles.root, {
      defaultVariants: {
        glass: false,
        shadow: false
      },
      /* eslint-disable sort-keys-fix/sort-keys-fix */
      variants: {
        glass: {
          false: null,
          true: styles.glass
        },
        shadow: {
          false: null,
          true: styles.shadow
        }
      }
      /* eslint-enable sort-keys-fix/sort-keys-fix */
    });
  }, [styles]);
  return /*#__PURE__*/_jsx(AntdButton, _objectSpread(_objectSpread({
    className: cx(variants({
      glass: glass,
      shadow: shadow
    }), className),
    color: color || (defaultVariant === 'filled' ? 'default' : undefined),
    danger: danger,
    icon: icon && ( /*#__PURE__*/isValidElement(icon) ? icon : /*#__PURE__*/_jsx(Icon, _objectSpread(_objectSpread({
      icon: icon
    }, iconProps), {}, {
      size: (iconProps === null || iconProps === void 0 ? void 0 : iconProps.size) || {
        size: '1.2em'
      }
    }))),
    loading: loading ? {
      icon: /*#__PURE__*/_jsx(Icon, _objectSpread(_objectSpread({
        icon: Loader2Icon
      }, iconProps), {}, {
        size: (iconProps === null || iconProps === void 0 ? void 0 : iconProps.size) || {
          size: '1.2em'
        },
        spin: true
      }))
    } : false,
    ref: ref,
    type: type,
    variant: defaultVariant
  }, rest), {}, {
    children: children
  }));
});
export default Button;
