// @lobehub/ui/es/styles/customTheme.js
import { blue, cyan, geekblue, gold, green, lime, magenta, orange, purple, red, volcano, yellow } from "../color/colors";
import { mauve, olive, sage, sand, slate } from "../color/neutrals";
export var primaryColors = {
  blue: blue.dark[9],
  cyan: cyan.dark[9],
  geekblue: geekblue.dark[9],
  gold: gold.dark[9],
  green: green.dark[9],
  lime: lime.dark[9],
  magenta: magenta.dark[9],
  orange: orange.dark[9],
  purple: purple.dark[9],
  red: red.dark[9],
  volcano: volcano.dark[9],
  yellow: yellow.dark[9]
};
export var primaryColorsSwatches = [primaryColors.red, primaryColors.orange, primaryColors.gold, primaryColors.yellow, primaryColors.lime, primaryColors.green, primaryColors.cyan, primaryColors.blue, primaryColors.geekblue, primaryColors.purple, primaryColors.magenta, primaryColors.volcano];
export var neutralColors = {
  mauve: mauve.dark[9],
  olive: olive.dark[9],
  sage: sage.dark[9],
  sand: sand.dark[9],
  slate: slate.dark[9]
};
export var neutralColorsSwatches = [neutralColors.mauve, neutralColors.slate, neutralColors.sage, neutralColors.olive, neutralColors.sand];
export var findCustomThemeName = function findCustomThemeName(type, value) {
  var res = type === 'primary' ? primaryColors : neutralColors;
  var result = Object.entries(res).find(function (item) {
    return item[1] === value;
  });
  return result === null || result === void 0 ? void 0 : result[0];
};
