// @lobehub/ui/es/Highlighter/Highlighter.js
'use client';

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
var _excluded = ["animated", "fullFeatured", "actionIconSize", "children", "language", "className", "copyable", "showLanguage", "variant", "shadow", "wrap", "bodyRender", "actionsRender", "enableTransformer", "theme", "icon", "fileName", "allowChangeLanguage", "defaultExpand"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import { cva } from 'class-variance-authority';
import { memo, useMemo, useState } from 'react';
import { Flexbox } from 'react-layout-kit';
import CopyButton from "../CopyButton";
import { getCodeLanguageDisplayName } from "./const";
import Tag from "../Tag";
import FullFeatured from "./FullFeatured";
import SyntaxHighlighter from "./SyntaxHighlighter";
import { useStyles } from "./style";
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
export var Highlighter = /*#__PURE__*/memo(function (_ref) {
  var animated = _ref.animated,
    fullFeatured = _ref.fullFeatured,
    actionIconSize = _ref.actionIconSize,
    children = _ref.children,
    _ref$language = _ref.language,
    language = _ref$language === void 0 ? 'markdown' : _ref$language,
    className = _ref.className,
    _ref$copyable = _ref.copyable,
    copyable = _ref$copyable === void 0 ? true : _ref$copyable,
    _ref$showLanguage = _ref.showLanguage,
    showLanguage = _ref$showLanguage === void 0 ? true : _ref$showLanguage,
    _ref$variant = _ref.variant,
    variant = _ref$variant === void 0 ? 'filled' : _ref$variant,
    shadow = _ref.shadow,
    wrap = _ref.wrap,
    bodyRender = _ref.bodyRender,
    actionsRender = _ref.actionsRender,
    enableTransformer = _ref.enableTransformer,
    theme = _ref.theme,
    icon = _ref.icon,
    fileName = _ref.fileName,
    allowChangeLanguage = _ref.allowChangeLanguage,
    _ref$defaultExpand = _ref.defaultExpand,
    defaultExpand = _ref$defaultExpand === void 0 ? true : _ref$defaultExpand,
    rest = _objectWithoutProperties(_ref, _excluded);
  var _useStyles = useStyles(),
    styles = _useStyles.styles,
    cx = _useStyles.cx;
  var _useState = useState(language),
    _useState2 = _slicedToArray(_useState, 2),
    lang = _useState2[0],
    setLang = _useState2[1];
  var variants = useMemo(function () {
    return cva(styles.root, {
      defaultVariants: {
        shadow: false,
        variant: 'filled',
        wrap: false
      },
      /* eslint-disable sort-keys-fix/sort-keys-fix */
      variants: {
        variant: {
          filled: styles.filled,
          outlined: styles.outlined,
          borderless: styles.borderless
        },
        shadow: {
          false: null,
          true: styles.shadow
        },
        wrap: {
          false: styles.nowrap,
          true: styles.wrap
        }
      }
      /* eslint-enable sort-keys-fix/sort-keys-fix */
    });
  }, [styles]);
  var tirmedChildren = children.trim();
  var originalActions = copyable && /*#__PURE__*/_jsx(CopyButton, {
    content: tirmedChildren,
    glass: true,
    size: actionIconSize
  });
  var actions = actionsRender ? actionsRender({
    actionIconSize: actionIconSize,
    content: tirmedChildren,
    language: lang,
    originalNode: originalActions
  }) : originalActions;
  var originalBody = /*#__PURE__*/_jsx(SyntaxHighlighter, {
    animated: animated,
    enableTransformer: enableTransformer,
    language: lang === null || lang === void 0 ? void 0 : lang.toLowerCase(),
    theme: theme,
    variant: variant,
    children: tirmedChildren
  });
  var displayName = useMemo(function () {
    if (fileName) return fileName;
    return getCodeLanguageDisplayName(language);
  }, [fileName, language]);
  var body = bodyRender ? bodyRender({
    content: tirmedChildren,
    language: lang,
    originalNode: originalBody
  }) : originalBody;
  if (fullFeatured) return /*#__PURE__*/_jsx(FullFeatured, _objectSpread(_objectSpread({
    actionsRender: actionsRender,
    allowChangeLanguage: allowChangeLanguage,
    className: className,
    content: tirmedChildren,
    copyable: copyable,
    defaultExpand: defaultExpand,
    fileName: fileName,
    icon: icon,
    language: language,
    setLanguage: setLang,
    shadow: shadow,
    showLanguage: showLanguage,
    variant: variant,
    wrap: wrap
  }, rest), {}, {
    children: body
  }));
  return /*#__PURE__*/_jsxs(Flexbox, _objectSpread(_objectSpread({
    className: cx(variants({
      shadow: shadow,
      variant: variant,
      wrap: wrap
    }), className),
    "data-code-type": "highlighter"
  }, rest), {}, {
    children: [/*#__PURE__*/_jsx(Flexbox, {
      align: 'center',
      className: styles.actions,
      flex: 'none',
      gap: 4,
      horizontal: true,
      children: actions
    }), showLanguage && language && /*#__PURE__*/_jsx(Tag, {
      className: styles.lang,
      children: displayName
    }), body]
  }));
});
Highlighter.displayName = 'Highlighter';
export default Highlighter;
