// @lobehub/ui/es/Layout/Layout.js
'use client';

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(r, l) { var t = null == r ? null : "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (null != t) { var e, n, i, u, a = [], f = !0, o = !1; try { if (i = (t = t.call(r)).next, 0 === l) { if (Object(t) !== t) return; f = !1; } else for (; !(f = (e = i.call(t)).done) && (a.push(e.value), a.length !== l); f = !0); } catch (r) { o = !0, n = r; } finally { try { if (!f && null != t.return && (u = t.return(), Object(u) !== u)) return; } finally { if (o) throw n; } } return a; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { useResponsive } from 'antd-style';
import { memo, useEffect, useState } from 'react';
import DraggablePanel from "../DraggablePanel";
import LayoutFooter from "./components/LayoutFooter";
import LayoutHeader from "./components/LayoutHeader";
import LayoutMain from "./components/LayoutMain";
import LayoutSidebar from "./components/LayoutSidebar";
import LayoutSidebarInner from "./components/LayoutSidebarInner";
import LayoutToc from "./components/LayoutToc";
import { useStyles } from "./style";
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
import { Fragment as _Fragment } from "react/jsx-runtime";
var Layout = /*#__PURE__*/memo(function (_ref) {
  var helmet = _ref.helmet,
    _ref$headerHeight = _ref.headerHeight,
    headerHeight = _ref$headerHeight === void 0 ? 64 : _ref$headerHeight,
    header = _ref.header,
    footer = _ref.footer,
    sidebar = _ref.sidebar,
    asideWidth = _ref.asideWidth,
    toc = _ref.toc,
    children = _ref.children,
    tocWidth = _ref.tocWidth;
  var _useStyles = useStyles(headerHeight),
    styles = _useStyles.styles;
  var _useResponsive = useResponsive(),
    mobile = _useResponsive.mobile,
    laptop = _useResponsive.laptop;
  var _useState = useState(true),
    _useState2 = _slicedToArray(_useState, 2),
    expand = _useState2[0],
    setExpand = _useState2[1];
  useEffect(function () {
    setExpand(Boolean(laptop));
  }, [laptop]);
  return /*#__PURE__*/_jsxs(_Fragment, {
    children: [helmet, header && /*#__PURE__*/_jsxs(LayoutHeader, {
      headerHeight: headerHeight,
      children: [header, mobile && toc && /*#__PURE__*/_jsx(LayoutToc, {
        children: toc
      })]
    }), /*#__PURE__*/_jsxs(LayoutMain, {
      children: [!mobile && !sidebar && /*#__PURE__*/_jsx("nav", {
        style: {
          width: tocWidth
        }
      }), !mobile && sidebar && /*#__PURE__*/_jsx(LayoutSidebar, {
        headerHeight: headerHeight,
        children: /*#__PURE__*/_jsx(DraggablePanel, {
          expand: expand,
          maxWidth: asideWidth,
          onExpandChange: setExpand,
          placement: "left",
          children: /*#__PURE__*/_jsx(LayoutSidebarInner, {
            headerHeight: headerHeight,
            children: sidebar
          })
        })
      }), /*#__PURE__*/_jsx("section", {
        className: styles.content,
        children: children
      }), !mobile && toc && /*#__PURE__*/_jsx(LayoutToc, {
        tocWidth: tocWidth,
        children: toc
      })]
    }), footer && /*#__PURE__*/_jsx(LayoutFooter, {
      children: footer
    })]
  });
});
export default Layout;
