// @lobehub/ui/es/Menu/Menu.js
'use client';

function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
var _excluded = ["compact", "shadow", "variant", "className", "selectable", "iconProps", "items", "ref"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import { Menu as AntdMenu, ConfigProvider } from 'antd';
import { cva } from 'class-variance-authority';
import { memo, useMemo } from 'react';
import { IconProvider } from "../Icon";
import { mapItems } from "./utils";
import { useStyles } from "./style";
import { jsx as _jsx } from "react/jsx-runtime";
var Menu = /*#__PURE__*/memo(function (_ref) {
  var compact = _ref.compact,
    shadow = _ref.shadow,
    _ref$variant = _ref.variant,
    variant = _ref$variant === void 0 ? 'borderless' : _ref$variant,
    className = _ref.className,
    selectable = _ref.selectable,
    iconProps = _ref.iconProps,
    items = _ref.items,
    ref = _ref.ref,
    rest = _objectWithoutProperties(_ref, _excluded);
  var _useStyles = useStyles(),
    cx = _useStyles.cx,
    styles = _useStyles.styles,
    theme = _useStyles.theme;
  var variants = useMemo(function () {
    return cva(styles.root, {
      defaultVariants: {
        compact: false,
        shadow: false,
        variant: 'borderless'
      },
      /* eslint-disable sort-keys-fix/sort-keys-fix */
      variants: {
        variant: {
          filled: styles.filled,
          outlined: styles.outlined,
          borderless: styles.borderless
        },
        shadow: {
          false: null,
          true: styles.shadow
        },
        compact: {
          false: null,
          true: styles.compact
        }
      }
      /* eslint-enable sort-keys-fix/sort-keys-fix */
    });
  }, [styles]);
  var antdItems = useMemo(function () {
    return items.map(function (item) {
      return mapItems(item);
    });
  }, [items]);
  return /*#__PURE__*/_jsx(ConfigProvider, {
    theme: {
      components: {
        Menu: {
          controlHeightLG: 36,
          iconMarginInlineEnd: 8,
          iconSize: 16,
          itemActiveBg: theme.isDarkMode ? theme.colorFillQuaternary : theme.colorFillSecondary,
          itemBorderRadius: theme.borderRadius,
          itemColor: theme.colorTextSecondary,
          itemHoverBg: theme.colorFillTertiary,
          itemMarginBlock: 4,
          itemMarginInline: 4,
          itemSelectedBg: theme.colorFillSecondary
        }
      }
    },
    children: /*#__PURE__*/_jsx(IconProvider, {
      config: _objectSpread({
        size: 'small'
      }, iconProps),
      children: /*#__PURE__*/_jsx(AntdMenu, _objectSpread({
        className: cx(variants({
          compact: compact,
          shadow: shadow,
          variant: variant
        }), className),
        inlineIndent: 12,
        items: antdItems,
        mode: "vertical",
        ref: ref,
        selectable: selectable
      }, rest))
    })
  });
});
Menu.displayName = 'Menu';
export default Menu;
