// @lobehub/ui/es/awesome/GridBackground/components/Grid.js
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
var _excluded = ["color", "strokeWidth", "linePick"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : String(i); }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
import { isUndefined } from 'lodash-es';
import { memo, useCallback } from 'react';
import { jsx as _jsx } from "react/jsx-runtime";
import { Fragment as _Fragment } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
var Line = /*#__PURE__*/function (Line) {
  Line[Line["l7"] = 0] = "l7";
  Line[Line["l6"] = 1] = "l6";
  Line[Line["l5"] = 2] = "l5";
  Line[Line["l4"] = 3] = "l4";
  Line[Line["l3"] = 4] = "l3";
  Line[Line["l2"] = 5] = "l2";
  Line[Line["l1"] = 6] = "l1";
  Line[Line["center"] = 7] = "center";
  Line[Line["r1"] = 8] = "r1";
  Line[Line["r2"] = 9] = "r2";
  Line[Line["r3"] = 10] = "r3";
  Line[Line["r4"] = 11] = "r4";
  Line[Line["r5"] = 12] = "r5";
  Line[Line["r6"] = 13] = "r6";
  Line[Line["r7"] = 14] = "r7";
  return Line;
}(Line || {});
var Grid = /*#__PURE__*/memo(function (_ref) {
  var _ref$color = _ref.color,
    color = _ref$color === void 0 ? '#fff' : _ref$color,
    _ref$strokeWidth = _ref.strokeWidth,
    strokeWidth = _ref$strokeWidth === void 0 ? 3 : _ref$strokeWidth,
    linePick = _ref.linePick,
    rest = _objectWithoutProperties(_ref, _excluded);
  var isUnpick = isUndefined(linePick);
  var showLine = useCallback(function (l) {
    return isUnpick || linePick === l;
  }, [linePick]);
  var vLine = /*#__PURE__*/_jsxs(_Fragment, {
    children: [showLine(Line.l7) && /*#__PURE__*/_jsx("path", {
      d: "M2 420v-60.343c0-21.82 14.15-41.12 34.959-47.684L1026 0h0"
    }), showLine(Line.l6) && /*#__PURE__*/_jsx("path", {
      d: "M268 420v-62.077c0-20.977 13.094-39.724 32.789-46.944L1149 0h0"
    }), showLine(Line.l5) && /*#__PURE__*/_jsx("path", {
      d: "M534 420v-64.358a50 50 0 0129.884-45.775L1269 0h0"
    }), showLine(Line.l4) && /*#__PURE__*/_jsx("path", {
      d: "M800 420v-67.395a50 50 0 0125.958-43.84L1389 0h0"
    }), showLine(Line.l3) && /*#__PURE__*/_jsx("path", {
      d: "M1066 420v-71.645a50 50 0 0120.456-40.337L1507 0h0"
    }), showLine(Line.l2) && /*#__PURE__*/_jsx("path", {
      d: "M1332 420v-77.506a50 50 0 0113.194-33.843L1629 0h0"
    }), showLine(Line.l1) && /*#__PURE__*/_jsx("path", {
      d: "M1598 420v-86.225a50 50 0 014.438-20.594L1744 0h0"
    }), showLine(Line.center) && /*#__PURE__*/_jsx("path", {
      d: "M1864 420V0h0"
    }), showLine(Line.r1) && /*#__PURE__*/_jsx("path", {
      d: "M2130 420v-86.225a50 50 0 00-4.438-20.594L1984 0h0"
    }), showLine(Line.r2) && /*#__PURE__*/_jsx("path", {
      d: "M2396 420v-77.506a50 50 0 00-13.194-33.843L2099 0h0"
    }), showLine(Line.r3) && /*#__PURE__*/_jsx("path", {
      d: "M2662 420v-71.645a50 50 0 00-20.456-40.337L2221 0h0"
    }), showLine(Line.r4) && /*#__PURE__*/_jsx("path", {
      d: "M2928 420v-67.395a50 50 0 00-25.958-43.84L2339 0h0"
    }), showLine(Line.r5) && /*#__PURE__*/_jsx("path", {
      d: "M3194 420v-64.358a50 50 0 00-29.884-45.775L2459 0h0"
    }), showLine(Line.r6) && /*#__PURE__*/_jsx("path", {
      d: "M3460 420v-62.077c0-20.977-13.094-39.724-32.789-46.944L2579 0h0"
    }), showLine(Line.r7) && /*#__PURE__*/_jsx("path", {
      d: "M3726 420v-60.343c0-21.82-14.15-41.12-34.959-47.684L2702 0h0"
    })]
  });
  var hLine = isUnpick && /*#__PURE__*/_jsxs(_Fragment, {
    children: [/*#__PURE__*/_jsx("path", {
      d: "M2835 42H892"
    }), /*#__PURE__*/_jsx("path", {
      d: "M595 136h2538"
    }), /*#__PURE__*/_jsx("path", {
      d: "M237 249h3254"
    })]
  });
  return /*#__PURE__*/_jsx("svg", _objectSpread(_objectSpread({
    style: {
      width: '100%'
    },
    viewBox: "0 0 3728 422",
    xmlns: "http://www.w3.org/2000/svg"
  }, rest), {}, {
    children: /*#__PURE__*/_jsxs("g", {
      fill: "none",
      fillRule: "evenodd",
      stroke: color,
      strokeWidth: strokeWidth,
      children: [vLine, hLine]
    })
  }));
});
Grid.displayName = 'Grid';
export default Grid;
