// @lobehub/ui/es/Block/style.js
var _templateObject, _templateObject2;
function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }
import { createStyles } from 'antd-style';
export var useStyles = createStyles(function (_ref) {
  var css = _ref.css,
    stylish = _ref.stylish,
    token = _ref.token;
  return {
    borderless: stylish.variantBorderlessWithoutHover,
    clickableBorderless: stylish.variantBorderless,
    clickableFilled: stylish.variantFilled,
    clickableOutlined: stylish.variantOutlined,
    clickableRoot: css(_templateObject || (_templateObject = _taggedTemplateLiteral(["\n      cursor: pointer;\n    "]))),
    filled: stylish.variantFilledWithoutHover,
    glass: stylish.blur,
    outlined: stylish.variantOutlinedWithoutHover,
    root: css(_templateObject2 || (_templateObject2 = _taggedTemplateLiteral(["\n      position: relative;\n      border-radius: ", "px;\n    "])), token.borderRadius),
    shadow: stylish.shadow
  };
});
